src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -0,0 +1,338 @@
+    """Action to synchronize input job attachments for a AWS Deadline Cloud job
+
+    Parameters
+    ----------
+    id : str
+        The unique action identifier
+    """
+
+    _cancel: Event
+    _future: Future[None]
+    _job_attachment_details: Optional[JobAttachmentDetails]
+    _step_details: Optional[StepDetails]
+    _step_script: Optional[StepScript_2023_09]
+
+    def __init__(
+        self,
+        *,
+        id: str,
+        session_id: str,
+        job_attachment_details: Optional[JobAttachmentDetails] = None,
+        step_details: Optional[StepDetails] = None,
+    ) -> None:
+        super(AttachmentDownloadAction, self).__init__(
+            id=id,
+            action_log_kind=(
+                SessionActionLogKind.JA_SYNC
+                if step_details is None
+                else SessionActionLogKind.JA_DEP_SYNC
+            ),
+            step_id=step_details.step_id if step_details is not None else None,
+        )
+        self._cancel = Event()
+        self._job_attachment_details = job_attachment_details
+        self._step_details = step_details
+        self._logger = LoggerAdapter(OPENJD_LOG, extra={"session_id": session_id})
+
+    def set_step_script(self, manifests, path_mapping, s3_settings) -> None:
+        profile = os.environ.get("AWS_PROFILE")
+        deadline_path = os.path.join(Path(sysconfig.get_path("scripts")), "deadline")
+
+        script = "#!/usr/bin/env bash\n\n{} attachment download -m {} --path-mapping-rules {} --s3-root-uri {}  --profile {}".format(
+            deadline_path,
+            " -m ".join(quote(v) for v in manifests),
+            quote(path_mapping),
+            s3_settings.to_s3_root_uri(),
+            profile,
+        )
+
+        self._step_script = StepScript_2023_09(
+            actions=StepActions_2023_09(
+                onRun=Action_2023_09(command="{{ Task.File.AttachmentDownload }}")
+            ),
+            embeddedFiles=[
+                EmbeddedFileText_2023_09(
+                    name="AttachmentDownload",
+                    type=EmbeddedFileTypes_2023_09.TEXT,
+                    runnable=True,
+                    data=script,
+                )
+            ],
+        )
+
+    def __eq__(self, other: Any) -> bool:
+        return (
+            type(self) is type(other)
+            and self._id == other._id
+            and self._job_attachment_details == other._job_attachment_details
+            and self._step_details == other._step_details
+        )
+
+    def start(
+        self,
+        *,
+        session: Session,
+        executor: Executor,
+    ) -> None:
+        """Initiates the synchronization of the input job attachments
+
+        Parameters
+        ----------
+        session : Session
+            The Session that is the target of the action
+        executor : Executor
+            An executor for running futures
+        """
+
+        self._logger.info(f"Syncing inputs using session {session}")
```

Comment:
The `openjd-sessions` library includes an `extra` in each log statement to classify the log record, for example:

https://github.com/OpenJobDescription/openjd-sessions-for-python/blob/fa8aa605c10222197961a7142a1fc73abe1e17b6/src/openjd/sessions/_action_filter.py#L159

This allows consumers of the logger to add conditional handling to the log records. I'm not certain, but we might want to do the same here to ensure the worker agent's log handling code doesn't omit these when deciding what to forward to CloudWatch.

Same feedback applies to the other logging statements below.

Reply:
Updated to add the extra info log content.